libs/perun/components/src/lib/groups-list/groups-list.component.ts

```diff
@@ -1,105 +1,126 @@
-    'name',
-    'status',
-    'groupStatus',
-    'description',
-    'expiration',
-    'menu',
-  ];
   @Input() disableMembers: boolean;
   @Input() disableGroups: boolean;
   @Input() groupsToDisableCheckbox: Set<number> = new Set<number>();
   @Input() groupsToDisableRouting: Set<number> = new Set<number>();
-  @Input() filter = '';
   @Input() disableHeadCheckbox: boolean;
   @Input() parentGroup: Group;
   @Input() disableRouting = false;
-  @Input() authType: string;
   @Input() memberId: number;
   @Input() memberGroupStatus: string;
   @Input() pageSizeOptions = TABLE_ITEMS_COUNT_OPTIONS;
   @Input() recentIds: number[] = [];
   @Input() resourceId: number = null;
   @Input() tableId: string;
   @Input() noGroupsAlert = 'SHARED_LIB.UI.ALERTS.NO_GROUPS';
+  @Output() groupMoved = new EventEmitter<GroupWithStatus>();
   @Output() refreshTable = new EventEmitter<void>();
-  @ViewChild(TableWrapperComponent, { static: true }) child: TableWrapperComponent;
+  @Output() queryChanged = new EventEmitter<PageQuery>();
+  @ViewChild(TableWrapperComponent, { static: true }) tableWrapper: TableWrapperComponent;
+  @ViewChild(MatSort, { static: true }) sort: MatSort;
 
   displayButtons = window.innerWidth > 800;
-  dataSource: MatTableDataSource<GroupWithStatus>;
+  dataSource: MatTableDataSource<GroupWithStatus> | DynamicDataSource<GroupWithStatus>;
   disabledRouting = false;
-  vo: Vo;
-  voIds: Set<number> = new Set<number>();
   voNames: Map<number, string> = new Map<number, string>();
-  removeAuth: boolean;
-
-  private sort: MatSort;
-  private hasMembersGroup = false;
+  columns: string[] = [
+    'select',
+    'id',
+    'recent',
+    'vo',
+    'indirectGroupAssigment',
+    'name',
+    'status',
+    'groupStatus',
+    'description',
+    'expiration',
+    'menu',
+  ];
 
   constructor(
     private dialog: MatDialog,
     private authResolver: GuiAuthResolver,
     private voService: VosManagerService,
     private tableCheckbox: TableCheckbox,
-    private changeDetector: ChangeDetectorRef
+    private disableGroupSelect: DisableGroupSelectPipe
   ) {}
 
-  @ViewChild(MatSort, { static: true }) set matSort(ms: MatSort) {
-    this.sort = ms;
+  @Input() set groups(groups: GroupWithStatus[] | PaginatedRichGroups) {
+    // Initialize data source with first group object passed
+    // One table instance can NOT alternate between paginated and not paginated groups
+    if (!this.dataSource) {
+      this.dataSourceInit(groups);
+    }
+
+    // Set up data correctly on each change
+    const paginated = this.isPaginated(groups);
+    if (isDynamicDataSource(this.dataSource) && paginated) {
+      this.dataSource.data = groups.data;
+      this.dataSource.count = groups.totalCount;
+    } else if (!isDynamicDataSource(this.dataSource) && !paginated) {
+      this.dataSource.data = groups;
+    }
+
+    this.updateVoNames();
+  }
+
+  @Input() set filter(value: string) {
+    this.dataSource.filter = value;
+  }
```

Comment:
Yes the IDE suggests this for some reason, not sure why. This si not the cause for the error. The error occures when filter input is defined before the group input. This leads me to believe that inputs are initialized in order they are passed in, rather than order they are defined. 
This order would cause an error since datasource will be initialized only after the groups are passed:
```
<perun-web-apps-groups-list
        [filter]="filterValue"
        [groups]="groups">
</perun-web-apps-groups-list>
```